deploy/terraform/main.tf

```diff
@@ -23,86 +23,86 @@ provider "metal" {
+  metro            = var.metro
   operating_system = "custom_ipxe"
   ipxe_script_url  = "https://boot.netboot.xyz"
   always_pxe       = "true"
   billing_cycle    = "hourly"
   project_id       = var.project_id
 }
 
-resource "metal_device_network_type" "tink_worker_network_type" {
-  device_id = metal_device.tink_worker.id
-  type      = "layer2-individual"
+resource "metal_port" "tink_worker_bond0" {
+  port_id = [for p in metal_device.tink_worker.ports : p.id if p.name == "bond0"][0]
+  layer2  = true
+  bonded  = false
+  #  vlan_ids = [metal_vlan.provisioning_vlan.id] 
+  # Can't do this: │ Error: vlan assignment batch could not be created: POST https://api.equinix.com/metal/v1/ports/b0bdf6d8-589e-4988-9000-9f49c97a54e1/vlan-assignments/batches: 422 Can't assign VLANs to port b0bdf6d8-589e-4988-9000-9f49c97a54e1, the port is configured for Layer 3 mode., Port b0bdf6d8-589e-4988-9000-9f49c97a54e1 cannot be assigned to VLANs., Bond disabled 
 }
 
 # Attach VLAN to worker
-resource "metal_port_vlan_attachment" "worker" {
-  depends_on = [metal_device_network_type.tink_worker_network_type]
-
-  device_id = metal_device.tink_worker.id
-  port_name = "eth0"
-  vlan_vnid = metal_vlan.provisioning_vlan.vxlan
+resource "metal_port" "tink_worker_eth0" {
+  depends_on = [metal_port.tink_worker_bond0]
+  port_id    = [for p in metal_device.tink_worker.ports : p.id if p.name == "eth0"][0]
+  #layer2     = true 
+  # TODO(displague) the terraform provider is not permitting this, perhaps a bug in the provider validation
+  # layer2 flag can be set only for bond ports
+  bonded   = false
+  vlan_ids = [metal_vlan.provisioning_vlan.id]
+  // vxlan_ids = [1000]
 }
 
-
 # Create a device and add it to tf_project_1
 resource "metal_device" "tink_provisioner" {
   hostname         = "tink-provisioner"
   plan             = var.device_type
-  facilities       = [var.facility]
+  metro            = var.metro
   operating_system = "ubuntu_20_04"
   billing_cycle    = "hourly"
   project_id       = var.project_id
-  user_data        = file("setup.sh")
+  user_data        = data.cloudinit_config.setup.rendered
 }
 
-resource "metal_device_network_type" "tink_provisioner_network_type" {
-  device_id = metal_device.tink_provisioner.id
-  type      = "hybrid"
+# Provisioners eth1 (unbonded) is attached to the provisioning VLAN
+resource "metal_port" "eth1" {
+  port_id  = [for p in metal_device.tink_provisioner.ports : p.id if p.name == "eth1"][0]
+  bonded   = false
+  vlan_ids = [metal_vlan.provisioning_vlan.id]
 }
 
-# Attach VLAN to provisioner
-resource "metal_port_vlan_attachment" "provisioner" {
-  depends_on = [metal_device_network_type.tink_provisioner_network_type]
-  device_id  = metal_device.tink_provisioner.id
-  port_name  = "eth1"
-  vlan_vnid  = metal_vlan.provisioning_vlan.vxlan
+data "archive_file" "compose" {
+  type        = "zip"
+  source_dir  = "${path.module}/../compose"
+  output_path = "${path.module}/compose.zip"
 }
```

Comment:
must we go with `zip` here to make this a module? Having to install unzip makes for a runtime failure (wan network issue...) vs using local-exec & tar the failure will happen early and solved "once".

Reply:
`zip` is what is available today in the Terraform module. Taking advantage of what is available in Terraform reduces our dependence on local compression tools.

https://registry.terraform.io/providers/hashicorp/archive/latest/docs/data-sources/archive_file 

https://github.com/hashicorp/terraform-provider-archive/pull/29

Reply:
We have other network dependencies that would fail with `zip`, be it package installs or Tink configuration.